File: test_api.js
async function test() {
  try {
    let res = await fetch('http://localhost:8000/api/v1/auth/login', {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({
        email: 'admin@stockpilot.dev',
        password: 'password123'
      })
    });
    let data = await res.json();
    if (!res.ok) {
        res = await fetch('http://localhost:8000/api/v1/auth/login', {
          method: 'POST',
          headers: { 'Content-Type': 'application/json' },
          body: JSON.stringify({
            email: 'test_user@example.com',
            password: 'password123'
          })
        });
        data = await res.json();
    }
    const token = data.access_token;
    console.log("Token acquired", token ? "Yes" : "No");
    
    // test post (sku omitted)
    let postRes = await fetch('http://localhost:8000/api/v1/products', {
      method: 'POST',
      headers: { 
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}` 
      },
      body: JSON.stringify({
        name: 'Test',
        price: 10,
        image_url: null
      })
    });
    let postData = await postRes.json();
    console.log("STATUS (omitted):", postRes.status);
    console.log("DATA (omitted):", JSON.stringify(postData, null, 2));

    console.log("--- TEST WITH SKU AS EMPTY STRING ---");
    let postRes2 = await fetch('http://localhost:8000/api/v1/products', {
      method: 'POST',
      headers: { 
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}` 
      },
      body: JSON.stringify({
        name: 'Test2',
        sku: "",
        price: 10,
        image_url: null
      })
    });
    let postData2 = await postRes2.json();
    console.log("STATUS (empty string):", postRes2.status);
    console.log("DATA (empty string):", JSON.stringify(postData2, null, 2));
    
  } catch (err) {
    console.log("ERROR:", err.message);
  }
}
test();

File: test_api.py
import urllib.request
import json
import ssl

def main():
    ctx = ssl.create_default_context()
    ctx.check_hostname = False
    ctx.verify_mode = ssl.CERT_NONE

    req = urllib.request.Request(
        "http://localhost:8000/api/v1/auth/login",
        data=json.dumps({"email": "test_user@example.com", "password": "password123"}).encode("utf-8"),
        headers={"Content-Type": "application/json"}
    )
    try:
        res = urllib.request.urlopen(req, context=ctx)
        token = json.loads(res.read())["access_token"]
        print("Logged in!")
    except Exception as e:
        print("Login failed:", e)
        try:
            print(e.read().decode())
        except:
            pass
        return

    headers = {
        "Authorization": f"Bearer {token}",
        "Content-Type": "application/json"
    }

    print("\n--- Test GET /customers ---")
    req1 = urllib.request.Request("http://localhost:8000/api/v1/customers", headers=headers)
    try:
        res1 = urllib.request.urlopen(req1, context=ctx)
        print(res1.status, res1.read()[:100])
    except Exception as e:
        print("Failed:", e)
        try:
            print(e.read().decode())
        except:
            pass
            
    print("\n--- Test GET /products ---")
    req2 = urllib.request.Request("http://localhost:8000/api/v1/products", headers=headers)
    try:
        res2 = urllib.request.urlopen(req2, context=ctx)
        print(res2.status, res2.read()[:100])
    except Exception as e:
        print("Failed:", e)
        try:
            print(e.read().decode())
        except:
            pass

if __name__ == "__main__":
    main()
